src/components/manners/implementation.ts

```diff
@@ -18,4 +31,19 @@ export type Implementation = {
    * By default this loads the required version of the wasm wnfs module from unpkg.com.
    */
   wnfsWasmLookup: (wnfsVersion: string) => Promise<BufferSource | Response>
+
+  /**
+   * File system.
+   */
+  fileSystem: {
+    /**
+     * Various file system hooks.
+     */
+    hooks: {
+      afterLoadExisting: (fs: FileSystem.API, dataFlowComponents: DataFlowComponents) => Promise<void>
+      afterLoadNew: (fs: FileSystem.API, dataFlowComponents: DataFlowComponents) => Promise<void>
+      beforeLoadExisting: (cid: CID, dataFlowComponents: DataFlowComponents) => Promise<void>
```

Comment:
Requirement for walletauth, I need to load part of the public filesystem to extract the encrypted root read key. I can't pass in the reference as a dependency during the creation of the manners components. because the reference component itself depends on the manners component. And sadly the `PublicTree` and `PrivateTree` components of the file system depend on the reference component, because they need to do DNS lookups. Should we add these hooks elsewhere? 🤷‍♂️